pkg/controller/controller_utils.go

```diff
@@ -481,38 +516,59 @@ type FakePodControl struct {
 	sync.Mutex
 	Templates     []api.PodTemplateSpec
 	DeletePodName []string
+	Patches       [][]byte
 	Err           error
 }
 
 var _ PodControlInterface = &FakePodControl{}
 
+func (f *FakePodControl) PatchPod(namespace, name string, data []byte) error {
+	f.Lock()
+	defer f.Unlock()
+	f.Patches = append(f.Patches, data)
+	if f.Err != nil {
+		return f.Err
+	}
+	return nil
+}
+
 func (f *FakePodControl) CreatePods(namespace string, spec *api.PodTemplateSpec, object runtime.Object) error {
 	f.Lock()
 	defer f.Unlock()
+	f.Templates = append(f.Templates, *spec)
 	if f.Err != nil {
 		return f.Err
 	}
+	return nil
+}
+
+func (f *FakePodControl) CreatePodsWithControllerRef(namespace string, spec *api.PodTemplateSpec, object runtime.Object, controllerGVK unversioned.GroupVersionKind) error {
+	f.Lock()
+	defer f.Unlock()
 	f.Templates = append(f.Templates, *spec)
```

Comment:
seems like this isn't a very good fake if it doesn't actually add the controller ref to the pod...